src/Sylius/Component/Core/Cart/Context/ShopBasedCartContext.php

```diff
@@ -76,6 +87,14 @@ private function setCustomerAndAddressOnCart(OrderInterface $cart, CustomerInter
     {
         $cart->setCustomer($customer);
 
+        if ($this->tokenStorage !== null) {
+            $user = $this->tokenStorage->getToken()->getUser();
+
+            if ($user !== null) {
+                $cart->setGuest(false);
+            }
+        }
+
```

Comment:
Hmmm... I'm sceptical 😄 I know we're doing some sets already in the context, but it still seems **bad** to set the value in the service that should just provide us with information. I'm wondering, maybe we can listen on user logging in and then changing this property on the current cart? And when the user logs out - change it again? It would be more natural, at least in my perspective 🖖